contracts/StorageManager.sol

```diff
@@ -46,48 +49,63 @@ contract StorageManager {
     // offerRegistry stores the open or closed Offer for provider.
     mapping(address => Offer) public offerRegistry;
 
+    // maps the tokenAddresses which can be used with this contract. By convention, address(0) is the native token.
+    mapping(address => bool) public isWhitelistedToken;
+
     event TotalCapacitySet(address indexed provider, uint128 capacity);
-    event BillingPlanSet(address indexed provider, uint64 period, uint64 price);
+    event BillingPlanSet(address indexed provider, address token, uint64 period, uint64 price);
     event MessageEmitted(address indexed provider, bytes32[] message);
 
     event NewAgreement(
-        bytes32 agreementReference,
-        bytes32[] dataReference,
+        bytes32[] dataReference,	
         address indexed agreementCreator,
         address indexed provider,
         uint128 size,
         uint64 billingPeriod,
         uint64 billingPrice,
-        uint256 availableFunds
+        uint256 availableFunds	
     );
-    event AgreementFundsDeposited(bytes32 indexed agreementReference, uint256 amount);
-    event AgreementFundsWithdrawn(bytes32 indexed agreementReference, uint256 amount);
-    event AgreementFundsPayout(bytes32 indexed agreementReference, uint256 amount);
+    event AgreementFundsDeposited(bytes32 indexed agreementReference, uint256 amount, address indexed token);
+    event AgreementFundsWithdrawn(bytes32 indexed agreementReference, uint256 amount, address indexed token);
+    event AgreementFundsPayout(bytes32 indexed agreementReference, uint256 amount, address indexed token);
     event AgreementStopped(bytes32 indexed agreementReference);
 
+    constructor() Ownable() public { }
+
+    /**
+    @notice whitelist a token or remove the token from whitelist
+    @param token the token from whom you want to set the whitelisted
+    @param isWhiteListed whether you want to whitelist the token or put it from the whitelist.
+    */
+    function setWhitelistedTokens(address token, bool isWhiteListed) public onlyOwner {
```

Comment:
Note: this also allows for disabling a token @julianmrodri

Reply:
Special attention is needed to verify that this doesn't create any edge conditions @vojtechsimetka

Reply:
> Note: this also allows for disabling a token @julianmrodri

I think its OK!